File: Editorial/2873_Maximum_Value_of_an_Ordered_Triplet_I/C/2_Greedy.c
long long maximumTripletValue(int* nums, int numsSize) {
    long long res = 0;
    for (int k = 2; k < numsSize; k++) {
        int maxPrefix = nums[0];
        for (int j = 1; j < k; j++) {
            long long value = (long long)(maxPrefix - nums[j]) * nums[k];
            if (value > res) {
                res = value;
            }
            if (nums[j] > maxPrefix) {
                maxPrefix = nums[j];
            }
        }
    }
    return res;
}

File: Editorial/2801-2900/2873_Maximum_Value_of_an_Ordered_Triplet_I/CSharp/2_Greedy.cs
public class Solution {
    public long MaximumTripletValue(int[] nums) {
        int n = nums.Length;
        long res = 0;
        for (int k = 2; k < n; k++) {
            int maxPrefix = nums[0];
            for (int j = 1; j < k; j++) {
                res = Math.Max(res, (long)(maxPrefix - nums[j]) * nums[k]);
                maxPrefix = Math.Max(maxPrefix, nums[j]);
            }
        }
        return res;
    }
}
